mod_auth/controllers.py

```diff
@@ -225,11 +226,17 @@ def login() -> Union[Response, Dict[str, Union[str, LoginForm]]]:
     """Route for handling the login page."""
     redirect_location = request.args.get('next', '')
 
+    # Validate redirect location to prevent open redirects
+    # Only allow paths starting with / (relative to this site)
+    if redirect_location and (not redirect_location.startswith('/') or redirect_location.startswith('//')):
+        redirect_location = ''
```

Comment:
This probably does not catch all cases... So I'd rather see the old version and just handle the error more gracefully.

Reply:
Agreed - reverted to the old approach. Now using try/except around url_for() to catch BuildError and redirect to home page instead. This avoids the security concerns while still preventing the 500 errors.